### PR TITLE
Add Ability to pass own client in AsyncHttpClient so that it's state can be referenced

### PR DESCRIPTION
**Pull request by harpresing**
Currently there is no way to access `org.asynchttpclient.DefaultAsyncHttpClient#getClientStat` when we create a `Resource[F, Client[F]]` using `org.http4s.client.asynchttpclient.AsyncHttpClient#resource`.

This PR adds an additional apply method in `AsyncHttpClient` which lets the user pass in their own `DefaultAsyncHttpClient` so that they can have a handle on its state and use it for metrics etc.

### CODE REVIEW

File: async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala
Comment (rossabaker) [inline async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala:198]: I'd put this class in its own source file.

Comment (harpresing) [reply async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala]: Moved to it's own file

Comment (harpresing) [reply async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala]: Makes sense, I've made it a simple class

File: async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala
Comment (harpresing) [reply async-http-client/src/main/scala/org/http4s/client/asynchttpclient/AsyncHttpClient.scala]: Done!

**Review by rossabaker — APPROVED** [review]
Test failure is unrelated. This looks great!

**Comment by harpresing** [comment]
Thanks for reviewing it @rossabaker! I can't merge it though since I don't have write access to the repository 😅

**Comment by rossabaker** [comment]
We usually aim for two reviewers, but sometimes our reviewers get busy.  I'll go ahead and merge it, and aim to release it this week.

**Pull request merged**
